ai/integrations/autogen/README.md

```diff
@@ -0,0 +1,281 @@
+
+#### Client setup
+
+Initialize a client for managing UC functions in a Databricks workspace, and set it as the global client.
+
+```python
+from unitycatalog.ai.core.client import set_uc_function_client
+from unitycatalog.ai.core.databricks import DatabricksFunctionClient
+
+client = DatabricksFunctionClient(
+    warehouse_id="...", # replace with the warehouse_id
+    cluster_id="..." # optional, only pass when you want to use cluster for function creation
+)
+
+# sets the default uc function client
+set_uc_function_client(client)
+```
+
+#### Create a function in UC
+
+Create Python UDFs in Unity Catalog with the client.
+
+```python
+# replace with your own catalog and schema
+CATALOG = "catalog"
+SCHEMA = "schema"
+
+from typing import Annotated
+
+def get_temperature(location: Annotated[str, "Retrieves the current weather from a provided location."]) -> str:
+    '''
+    Returns the current temperature from a given location in degrees Celsius.
+    '''
+    return "31.9 C"
+
+def temp_c_to_f(celsius: Annotated[str, "Temperature in Celsius"]) -> float:
+    '''
+    Converts temperature from Celsius to Fahrenheit.
+    '''
+    celsius = float(celsius)
+    fahrenheit = (9/5 * celsius) + 32
+    return fahrenheit
+
+# Create UC functions
+function_info_temp_c_to_f = client.create_python_function(
+    func=TempCtoF,
+    catalog=CATALOG,
+    schema=SCHEMA
+)
+
+function_info_get_temp = client.create_python_function(
+    func=get_temperature,
+    catalog=CATALOG,
+    schema=SCHEMA
+)
+```
+
+Now that the functions are created and stored in the corresponding catalog and schema, we can use it within autogen's SDK.
+
+#### Create a UCFunctionToolkit instance
+
+To begin, we will need an instance of the tool function interface from the `unitycatalog_autogen` toolkit.
+
+```python
+from unitycatalog.ai.autogen.toolkit import UCFunctionToolkit
+
+# Create an instance of the toolkit with the functions that were created earlier.
+# Use the full function names in 'catalog.schema.function_name' format
+function_names = [
+    f"{CATALOG}.{SCHEMA}.get_temperature",
+    f"{CATALOG}.{SCHEMA}.TempCtoF"]
+
+toolkit = UCFunctionToolkit(function_names=function_names,
+                            client=client)
+
+# Access the tool definitions that are in the interface that autogen's SDK expects
+tools = toolkit.tools
+
+```
+
+Now that we have the defined tools from Unity Catalog, we can directly pass this definition to the Conversable Agent API within Autogen.
+
+If you would like to validate that your tool is functional prior to proceeding to integrate it with Autogen, you can call the tool directly:
+
+```python
+my_tool = tools[0]
+
+my_tool.fn(**{"location": "San Francisco"})
+```
+
+#### Use the tools with a conversable Agent
+
+```python
+
+import os
+from autogen import ConversableAgent, GroupChat, GroupChatManager
+
+OPENAI_API_KEY = '.....'
```

Comment:
Nit: can we leverage environment variables instead?

Reply:
Done